src/com/sun/jna/NativeLibrary.java

```diff
@@ -945,4 +999,23 @@ else if (Platform.isARM()) {
         }
         return ldPaths;
     }
+
+    /**
+     * Expand the given search path with the given suffixes.
+     */
+    private static List expandPathsWithSuffix(List searchPath, List suffixes) {
+        List suffixSearchPath = new LinkedList();
+        for (int i=0; i < searchPath.size(); i++) {
+        	String nextPath = (String) searchPath.get(i);
+        	suffixSearchPath.add(nextPath);
+        	for (int j=0; j < suffixes.size(); j++) {
+        		String suffix = (String) suffixes.get(j);
+        		File f = new File(nextPath, suffix);
+        		suffixSearchPath.add(f.getPath());
+        	}
+        }
+
+        return suffixSearchPath;
+    }
+
```

Comment:
Let's use generics since we are using Java 1.5+ and then we can remove the explicit casts